au4/src/projectscene/qml/Audacity/ProjectScene/trackspanel/EditableLabel.qml

```diff
@@ -0,0 +1,98 @@
+import QtQuick
+import QtQuick.Controls
+import QtQuick.Layouts
+
+import Muse.Ui
+import Muse.UiComponents
+
+Item {
+    id: root
+
+    Layout.fillWidth: true
+    Layout.fillHeight: true
+
+    property alias text: label.text
+    property alias font: label.font
+
+    signal textEdited(string text)
+
+    function edit() {
+        editLoader.edit(label.text)
+    }
+
+    MouseArea {
+        anchors.fill: parent
+
+        onDoubleClicked: {
+            editLoader.edit(label.text)
+        }
+    }
+
+    StyledTextLabel {
+        id: label
+
+        anchors.fill: parent
+
+        font: ui.theme.bodyFont
+        horizontalAlignment: Text.AlignLeft
+    }
+
+    Loader {
+        id: editLoader
+
+        anchors.fill: title
+
+        property bool isEditState: false
+
+        sourceComponent: editLoader.isEditState ? textEditComp : null
```

Comment:
I would suggest to put StyledTextLabel inside Loader and use it instead of null. Then it should be like this: sourceComponent: editLoader.isEditState ? textEditComp : textLabelComp